src/components/TextInput/TextInput.tsx

```diff
@@ -15,29 +17,38 @@ export type TextInputProps = NativeAttributes<'input'> & {
 
   /** Whether the input is disabled or not */
   disabled?: boolean;
+
+  /** The icon present on the input  */
+  icon?: IconProps['type'];
 };
 
 const TextInput = React.forwardRef<HTMLInputElement, TextInputProps>(function TextInput(
-  { label, invalid, required, disabled, id, hidden, name, value, ...rest },
+  { label, invalid, required, disabled, id, hidden, name, icon, value, ...rest },
   ref
 ) {
   const identifier = id || name || slugify(label);
 
   return (
     <InputControl invalid={invalid} disabled={disabled} required={required} hidden={hidden}>
-      <InputElement
-        as="input"
-        type="text"
-        id={identifier}
-        name={name}
-        value={value}
-        truncated
-        {...rest}
-        ref={ref}
-      />
-      <InputLabel raised={!!value} htmlFor={identifier}>
-        {label}
-      </InputLabel>
+      <Flex align="center" position="relative">
+        <InputElement
+          as="input"
+          type="text"
+          id={identifier}
+          name={name}
+          value={value}
+          truncated
+          pr={icon && 40}
```

Comment:
For future reference, we use a `4px` scale, thus `10`  = `40px`. 

Now, the scale goes from `0` up to `10`. Anything above it is **automatically** translated into `px` as a fallback. Thus in your case, using either `10` or `40` would be the same. This is because:

- 0 = 0
- 1 = 4px
- 2 = 8px
....
- 10 = 40px

**SCALE STOPS**

- 11 = 11px
- 12 = 12px
....
- 40 = 40px

Similarly: